jupyterlab_automl/jupyterlab_automl/handlers.py

```diff
@@ -60,109 +95,141 @@ def create_automl_parent(client):
+        return [month_chart, day_chart, time_chart]
+    else:
+        return []
+
+
+def get_column_specs(client, table_spec):
     column_specs = []
-    types = {
-        0: "Unspecified",
-        3: "Numeric",
-        4: "Timestamp",
-        6: "String",
-        8: "Array",
-        9: "Struct",
-        10: "Categorical",
-    }
-    summary = defaultdict(int)
-    for column_spec in client.list_column_specs(tableSpecId):
-        chart_info = []
-        if column_spec.data_type.type_code in types.keys():
-            type_code = types[column_spec.data_type.type_code]
-            if column_spec.data_type.type_code == 3:
-                chart_data = []
-                mean = column_spec.data_stats.float64_stats.mean
-                standard_deviation = column_spec.data_stats.float64_stats.standard_deviation
-                for bucket in column_spec.data_stats.float64_stats.histogram_buckets:
-                    if bucket.min == float("-inf"):
-                        label = "[" + str(bucket.min) + ", " + str(round(bucket.max)) + "]"
-                    elif bucket.max == float("inf"):
-                        label = "[" + str(round(bucket.min)) + ", " + str(bucket.max) + "]"
-                    else: 
-                        label = "[" + str(round(bucket.min)) + ", " + str(round(bucket.max)) + "]"
-                    chart_data.append({"name": label, "amount": bucket.count})
-                chart_info = [mean, standard_deviation, chart_data]
-            elif column_spec.data_type.type_code == 10:
-                for stat in column_spec.data_stats.category_stats.top_category_stats:
-                    chart_info.append({"name":stat.value, "amount": stat.count})
+    type_summary = defaultdict(int)
+    for column_spec in client.list_column_specs(table_spec.name):
+        if column_spec.data_type.type_code in column_types.keys():
+            type_code = column_types[column_spec.data_type.type_code]
         else:
             type_code = "Unrecognized"
-        summary[type_code] += 1
+        detail_panel = get_detail_panel(column_spec, table_spec.row_count - column_spec.data_stats.null_value_count)
+        type_summary[type_code] += 1
         column_specs.append(
             {
                 "id": column_spec.name,
                 "dataType": type_code,
                 "displayName": column_spec.display_name,
                 "distinctValueCount": column_spec.data_stats.distinct_value_count,
-                "invalidValueCount": rowCount - column_spec.data_stats.valid_value_count,
-                "nullValueCount": str(column_spec.data_stats.null_value_count) + " (" + str(math.floor(column_spec.data_stats.null_value_count / rowCount * 100)) + "%)",
+                "invalidValueCount": table_spec.row_count
+                - column_spec.data_stats.valid_value_count,
+                "nullValueCount": str(column_spec.data_stats.null_value_count)
```

Comment:
why not have this in the Typescript code? the api returned a preformatted string could be confusing if we think it should return a number